stdlib/public/Concurrency/TaskPrivate.h

```diff
@@ -102,35 +124,36 @@ static void runTaskWithFutureResult(
 }
 
 /// Run the given task, providing it with the result of the future.
-static void runTaskWithGroupPollResult(
+static void runTaskWithPollResult(
     AsyncTask *waitingTask, ExecutorRef executor,
-    AsyncTask::GroupFragment::GroupPollResult result) {
+    TaskGroup::PollResult result) {
   auto waitingTaskContext =
       static_cast<TaskFutureWaitAsyncContext *>(waitingTask->ResumeContext);
 
   // Was it an error or successful return?
   waitingTaskContext->result.hadErrorResult =
-      result.status == AsyncTask::GroupFragment::GroupPollStatus::Error;
+      result.status == TaskGroup::GroupPollStatus::Error;
 
   // Extract the stored value into the waiting task's result storage:
   switch (result.status) {
-    case AsyncTask::GroupFragment::GroupPollStatus::Success:
+    case TaskGroup::GroupPollStatus::Success:
       waitingTaskContext->result.storage = result.storage;
       break;
-    case AsyncTask::GroupFragment::GroupPollStatus::Error:
+    case TaskGroup::GroupPollStatus::Error:
       waitingTaskContext->result.storage =
         reinterpret_cast<OpaqueValue *>(result.storage);
       break;
-    case AsyncTask::GroupFragment::GroupPollStatus::Empty:
+    case TaskGroup::GroupPollStatus::Empty:
       // return a `nil` here (as result of the `group.next()`)
       waitingTaskContext->result.storage = nullptr;
       break;
-    case AsyncTask::GroupFragment::GroupPollStatus::Waiting:
+    case TaskGroup::GroupPollStatus::Waiting:
       assert(false && "Must not attempt to run with a Waiting result.");
   }
 
   // TODO: schedule this task on the executor rather than running it directly.
-  waitingTask->run(executor);
+  // FIXME: or   waitingTask->run(executor); ?
+  waitingTask->run(waitingTaskContext->ResumeParentExecutor);
```

Comment:
TODO: please sanity check here... I think when we resume the waiting task we should resume it on the executor it has set when suspending... so this is right now?

We can revisit this as we allow customizing executors.